pkg/lib/cockpit-components-multi-typeahead-select.tsx

```diff
@@ -0,0 +1,390 @@
+
+   - We use Labels instead of Chips, since we want colors.
+
+   - The clear button clears the input text, not the selection.
+
+*/
+
+/* eslint-disable */
+
+import cockpit from "cockpit";
+import React from 'react';
+import {
+  Select,
+  SelectOption,
+  SelectList,
+  SelectOptionProps,
+  MenuToggle,
+  MenuToggleElement,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+  Button,
+  MenuToggleProps,
+  SelectProps,
+} from '@patternfly/react-core';
+import { Label, LabelGroup, LabelProps } from "@patternfly/react-core/dist/esm/components/Label/index.js";
+import TimesIcon from '@patternfly/react-icons/dist/esm/icons/times-icon';
+
+const _ = cockpit.gettext;
+
+export interface MultiTypeaheadSelectOption extends Omit<SelectOptionProps, 'content' | 'isSelected'> {
+  /** Content of the select option. */
+  content: string | number;
+  /** Value of the select option. */
+  value: string | number;
+  /** Color */
+  color?: LabelProps["color"];
+}
+
+export interface MultiTypeaheadSelectProps extends Omit<SelectProps, 'toggle' | 'onSelect'> {
+  /** @hide Forwarded ref */
+  innerRef?: React.Ref<any>;
+  /** Options of the select. */
+  options: MultiTypeaheadSelectOption[];
+  /** Selected values */
+  selected: (string | number)[];
+  /** Callback triggered when an option is added. */
+  onAdd: (value: (string | number)) => void;
+  /** Callback triggered wehn an option is removed. */
+  onRemove: (value: (string | number)) => void;
+  /** Callback triggered when the select opens or closes. */
+  onToggle?: (nextIsOpen: boolean) => void;
+  /** Callback triggered when the text in the input field changes. */
+  onInputChange?: (newValue: string) => void;
+  /** Placeholder text for the select input. */
+  placeholder?: string;
+  /** Message to display when no options match the filter. */
+  noOptionsFoundMessage?: string | ((filter: string) => string);
+  /** Flag indicating the select should be disabled. */
+  isDisabled?: boolean;
+  /** Width of the toggle. */
+  toggleWidth?: string;
+  /** Additional props passed to the toggle. */
+  toggleProps?: MenuToggleProps;
+}
+
+export const MultiTypeaheadSelectBase: React.FunctionComponent<MultiTypeaheadSelectProps> = ({
+  innerRef,
+  options,
+  selected,
+  onAdd,
+  onRemove,
+  onToggle,
+  onInputChange,
+  placeholder = '',
+  noOptionsFoundMessage = _filter => _("No results found"),
+  isDisabled = false,
+  toggleWidth,
+  toggleProps,
+  ...props
+}: MultiTypeaheadSelectProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [inputValue, setInputValue] = React.useState<string>("");
+  const [selectOptions, setSelectOptions] = React.useState<MultiTypeaheadSelectOption[]>(options);
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(null);
+  const [activeItemId, setActiveItemId] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+
+  const NO_RESULTS = 'no results';
```

Comment:
Needs translation

Reply:
This never shows up in the UI, it's a weird symbolic value.

Reply:
To elaborate a bit: The options each have a `value` and a `content`. The content shows up in the UI, for example "Automatically via NTP", the value is used by the program as a symbolic string, such as "ntp".  The `NO_RESULTS` string here is used as the `value` of the artificial "No results found" option when filtering doesn't find anything. It will never be returned to the user, and never show up in the UI. 

(It's junior code, of course, and maybe we should fix it.)